regression/library/test_manycore_alignment.c

```diff
@@ -0,0 +1,327 @@
+// 
+// Redistribution and use in source and binary forms, with or without modification,
+// are permitted provided that the following conditions are met:
+// 
+// Redistributions of source code must retain the above copyright notice, this list
+// of conditions and the following disclaimer.
+// 
+// Redistributions in binary form must reproduce the above copyright notice, this
+// list of conditions and the following disclaimer in the documentation and/or
+// other materials provided with the distribution.
+// 
+// Neither the name of the copyright holder nor the names of its contributors may
+// be used to endorse or promote products derived from this software without
+// specific prior written permission.
+// 
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+// ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+// WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR
+// ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+// (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+// LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+// ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#include <inttypes.h>
+#include <bsg_manycore.h>
+#include <bsg_manycore_config.h>
+#include <bsg_manycore_coordinate.h>
+#include <bsg_manycore_printing.h>
+#include <bsg_manycore_npa.h>
+#include "test_manycore_alignment.h"
+
+#define TEST_NAME "test_manycore_alignment"
+
+#define BASE_ADDR 0x0000
+
+int test_manycore_alignment() {
+        /********/
+        /* INIT */
+        /********/
+        int err, r = HB_MC_FAIL;
+        hb_mc_manycore_t manycore = {0}, *mc = &manycore;
+
+        srand(time(0));
+        
+        err = hb_mc_manycore_init(mc, TEST_NAME, 0);
+        if (err != HB_MC_SUCCESS) {
+                bsg_pr_err("%s: failed to intialize manycore: %s\n",
+                           __func__,
+                           hb_mc_strerror(err));
+                return HB_MC_FAIL;
+        }
+
+        const hb_mc_config_t *config = hb_mc_manycore_get_config(mc);
+        uint32_t manycore_dim_x = hb_mc_coordinate_get_x(hb_mc_config_get_dimension_vcore(config));
+
+        hb_mc_dimension_t dim = hb_mc_config_get_dimension_network(config);
+
+        uint32_t dram_coord_x = 0;
+
+        uint32_t dram_coord_y = hb_mc_config_get_dram_y(config);
+        int mismatch = 0;
+
+        /*
+         * Loop over all DRAM banks and try to read/write to an unaligned address
+         */
+        for (dram_coord_x = 0; dram_coord_x < hb_mc_dimension_get_x(dim); dram_coord_x++) {
```

Comment:
Even though these are all supposed to return error, some random testing would be good.

Reply:
As in random addresses. I can see that the data is random.